src/transformers/generation/configuration_utils.py

```diff
@@ -332,12 +332,121 @@ def __repr__(self):
 
     def validate(self):
         """
-        Validates the values of the attributes of the GenerationConfig instance, and raises a `ValueError` if any of
-        the values are invalid.
+        Validates the values of the attributes of the [`GenerationConfig`] instance. Raises exceptions in the presence
+        of parameterization that can be detected as incorrect from the configuration instance alone.
+
+        Note that some parameters are best validated at generate runtime, as they may depend on other inputs and/or the
+        model, such as parameters related to the generation length.
         """
+
+        # Validation of individual attributes
         if self.early_stopping not in {True, False, "never"}:
             raise ValueError(f"`early_stopping` must be a boolean or 'never', but is {self.early_stopping}.")
 
+        # Validation of attribute relations:
+        # 1. detect sampling-only parameterization when not in sampling mode
```

Comment:
The validation in `1.` and `2.` is new, `3.` and `4.` were moved from `generate` (as it falls under `generation_config` level validation)

The new validation checks throw exceptions, as they correspond to an impossible combination of arguments (and most users ignore warnings). Happy to lower them to warnings, if you prefer to optimize for backwards compatibility :)